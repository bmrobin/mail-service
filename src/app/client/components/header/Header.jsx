import React from 'react';
import { Jumbotron, Grid, Row, Col, Nav, NavItem } from 'react-bootstrap';
import img from './mail_icon.png';
import './header.css';

class Header extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      contactListSelected: true,
      newContactSelected: false
    };
  }

  componentDidMount() {
    fetch('http://localhost:9000/users', {
      method: 'GET',
      mode: 'cors'
    }).then((result) => {
        return result.json();
      })
      .then((data) => {
        console.log(data);
      })
      .catch((error) => {
        console.error(error);
      });
  }

  navSelected(event) {
    switch (event) {
      case 'list':
        this.setState({ contactListSelected: true, newContactSelected: false });
        break;
      case 'new':
        this.setState({ newContactSelected: true, contactListSelected: false });
        break;
    }
  }

  render() {
    return (
      <div>
        <Jumbotron>
          <img src={img} className="logo"></img>
          <h1 className="header-label">mail-service&nbsp;|<span className="description">&nbsp;&nbsp;A mail service for you</span></h1>
        </Jumbotron>

        <Grid>
          <Row>
            <Col xs={12} md={8}>
              <Nav bsStyle="tabs" onSelect={(event) => this.navSelected(event)}>
                <NavItem eventKey="list" active={this.state.contactListSelected}>Contact List</NavItem>
                <NavItem eventKey="new" active={this.state.newContactSelected}>New Contact</NavItem>
              </Nav>
            </Col>
          </Row>
        </Grid>
      </div>
    );
  }
}

export default Header;
