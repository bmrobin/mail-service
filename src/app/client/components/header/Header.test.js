import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

describe("Header Component", () => {

  test("should render component", () => {
    const component = renderer.create(<Header />).toJSON();
    expect(component).toMatchSnapshot();
  });

});
