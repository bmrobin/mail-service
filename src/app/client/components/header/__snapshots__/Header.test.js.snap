// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render component 1`] = `
<div>
  <div
    className="jumbotron"
  >
    <img
      className="logo"
      src="test-file-stub"
    />
    <h1
      className="header-label"
    >
      mail-service |
      <span
        className="description"
      >
          A mail service for you
      </span>
    </h1>
    <div
      className="github-logo"
    >
      <a
        href="https://github.com/bmrobin/mail-service/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          src="test-file-stub"
        />
      </a>
    </div>
  </div>
</div>
`;
